internal/translation/teams/teams.go

```diff
@@ -0,0 +1,130 @@
+package teams
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+)
+
+type AtlasTeamsService interface {
+	ListProjectTeams(ctx context.Context, projectID string) ([]Team, error)
+	GetTeamByName(ctx context.Context, orgID, teamName string) (*Team, error)
+	GetTeamByID(ctx context.Context, orgID, teamID string) (*Team, error)
+	Assign(ctx context.Context, at *[]Team, projectID string) error
+	Unassign(ctx context.Context, projectID, teamID string) error
+	Create(ctx context.Context, at *AssignedTeam, orgID string) (*Team, error)
+
+	GetTeamUsers(ctx context.Context, orgID, teamID string) ([]TeamUser, error)
+	UpdateRoles(ctx context.Context, at *Team, projectID string, newRoles []akov2.TeamRole) error
+	AddUsers(ctx context.Context, usersToAdd *[]admin.AddUserToTeam, orgID, teamID string) error
+	RemoveUser(ctx context.Context, orgID, teamID, userID string) error
+	RenameTeam(ctx context.Context, at *Team, orgID, newName string) (*Team, error)
+}
```

Comment:
I would reorder a bit for clarity on type managed:
```suggestion
type AtlasTeamsService interface {
	// Team  
	ListProjectTeams(ctx context.Context, projectID string) ([]Team, error)
	GetTeamByName(ctx context.Context, orgID, teamName string) (*Team, error)
	GetTeamByID(ctx context.Context, orgID, teamID string) (*Team, error)
	Assign(ctx context.Context, at *[]Team, projectID string) error
	Unassign(ctx context.Context, projectID, teamID string) error
	Create(ctx context.Context, at *AssignedTeam, orgID string) (*Team, error)
	UpdateRoles(ctx context.Context, at *Team, projectID string, newRoles []akov2.TeamRole) error
	RenameTeam(ctx context.Context, at *Team, orgID, newName string) (*Team, error)

	// TeamUser
	GetTeamUsers(ctx context.Context, orgID, teamID string) ([]TeamUser, error)
	AddUsers(ctx context.Context, usersToAdd *[]admin.AddUserToTeam, orgID, teamID string) error
	RemoveUser(ctx context.Context, orgID, teamID, userID string) error
}
```

Or maybe even separate interfaces joined by embedding:
```go
type AtlasTeamsService interface {
    TeamService
    TeamUserService
}
```